src/docmaker.ts

```diff
@@ -80,12 +77,12 @@ const run = async () => {
     ...(await loadData(config.data))
   };
 
+  const processor = new FileProcessor();
   const renderer = new FileRenderer(data);
 
   // Render all pages with data & join content
-  const content = (
-    await Promise.all(config.pages.map(page => renderer.renderFile(page)))
-  ).join("");
+  const joinedFiles = await processor.joinFiles(config.pages);
```

Comment:
See my comment in the FileProcessor.ts, I don't think this should be a class, or at least `joinFiles` should be a static method.